src/eval/execute.rs

```diff
@@ -95,6 +95,14 @@ fn generate_code_to_send(code: &str, bare: bool) -> String {
             code = body
         ))
     };
+
+    // Normalize the code if `raw` is false.
+    let code = if raw {
```

Comment:
This doesn't feel like the right place for this code. It is after some code wrapping, but before others, and there is no compelling reason why they are different and should be treated differently.

 I can see several issues putting it here:
 * the header part is not processed
 * when using `bare`, the code is not processed at all, which is probably undesirable given that `bare` and `raw` are orthogonal

I would suggest that we put the code into `run_code`, before calling `generate_code_to_send`. The latter then serves the only purpose to wrap the code provided. It would also avoid passing another boolean flag into the function.